ESP32/test/test_embedded/test_wifi.cpp

```diff
@@ -0,0 +1,56 @@
+#include "test_wifi.h"
+
+// void test_wifi_default_connection(void) {
+TEST_F(WifiTest, DefaultConnectionEstablish) {
+  esp_err_t err_connection;
+
+  // connect to wifi defined in the credentials
+  err_connection = m_wifi_handler.connect_to_wifi();
+
+  EXPECT_EQ(err_connection, ESP_OK)
+      << "Wifi connection could not be established";
+}
+
+TEST_F(WifiTest, Scan) {
+  esp_err_t err_connection;
+
+  err_connection = m_wifi_handler.connect_to_wifi();
+
+  EXPECT_EQ(err_connection, ESP_OK)
```

Comment:
explicit definition of err_connection not necessary, could all be done inside expect